VietNQ/ex4/ex4.ts

```diff
@@ -0,0 +1,69 @@
+/*
+
+Intro:
+
+    As we introduced "type" to both User and Admin
+    it's now easier to distinguish between them.
+    Once object type checking logic was extracted
+    into separate functions isUser and isAdmin -
+    logPerson function got new type errors.
+
+Exercise:
+
+    Figure out how to help TypeScript understand types in
+    this situation and apply necessary fixes.
+
+*/
+
+interface User {
+  type: 'user';
+  name: string;
+  age: number;
+  occupation: string;
+}
+
+interface Admin {
+  type: 'admin';
+  name: string;
+  age: number;
+  role: string;
+}
+
+export type Person = User | Admin;
+
+export const persons: Person[] = [
+  { type: 'user', name: 'Max Mustermann', age: 25, occupation: 'Chimney sweep' },
+  { type: 'admin', name: 'Jane Doe', age: 32, role: 'Administrator' },
+  { type: 'user', name: 'Kate Müller', age: 23, occupation: 'Astronaut' },
+  { type: 'admin', name: 'Bruce Willis', age: 64, role: 'World saver' }
+];
+
+export function isAdmin(person: Person) {
```

Comment:
hoặc k sửa ở dưới mà sửa ntn xem
```suggestion
export function isAdmin(person: Person): person is Admin {
```